overlord/state/taskrunner.go

```diff
@@ -84,22 +90,26 @@ func (r *TaskRunner) run(fn HandlerFunc, task *Task) {
 	tomb := &tomb.Tomb{}
 	r.tombs[task.ID()] = tomb
 	tomb.Go(func() error {
-		err := fn(task, tomb)
+		tomb.Kill(fn(task, tomb))
```

Comment:
Maybe a small code comment here what kill is doing? If you look at this without too much tomb knowledge the pattern `tomb.Go(.func() error {tomb.Kill())` looks a bit odd. But maybe that is just me, so feel free to ignore if its actually obvious and/or a common tomb pattern.